tests/integration/src/felt_conversion.rs

```diff
@@ -0,0 +1,117 @@
+use miden_core::Felt;
+use miden_core::StarkField;
+
+/// Wrapper around `Felt` that implements `From` for a bunch of types that are missing in Winterfell
+/// Should be moved to Winterfell eventually
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct TestFelt(pub Felt);
```

Comment:
So we'll have to use `TryFrom` for the signed types, since technically we can't represent negative values in a `Felt`. Obviously in practice we're encoding negative values by casting to unsigned, but I doubt the `winterfell` folks are going to be ok with merging conversions like that.

Unfortunately, that means that for signed types, we're going to simply be forced to either maintain a wrapper like this, or use `Felt::new(some_i32 as u32 as u64)` to handle representing signed types in general.

Reply:
I'd stick with this wrapper and don't try to merge them into the `winterfell` since we need a uniform way of going from Rust types to `Felt` and back for all integral types we want to use in `instruction` tests. See https://github.com/0xPolygonMiden/miden-ir/blob/4496a697eb98798ed41ff53a87f51e0ae1bcf336/tests/integration/src/rust_masm_tests/instructions.rs#L90-L151 . Macros are written to be type-agnostic and work for any passed type. Any supported Rust type is handled the same way in the macros.
The doc string is misleading since I wrote it before discovering the above requirement.

Reply:
That's fine, might be worth removing that comment in the source then. I guess some of the conversions might still be worth upstreaming, but we don't need to make any effort to do that IMO.